### PR TITLE
Day 2

### PR DESCRIPTION
**Pull request by tomalley104**
@alanscarpa 👀

### CODE REVIEW

File: twitterMini/NSDate+TimeFrom.swift [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+//  twitterMini
+//
+//  Created by Tom O'Malley on 5/3/16.
+//  Copyright © 2016 intrepid. All rights reserved.
+//
+
+import Foundation
+
+extension NSDate {
+    func yearsFrom(date:NSDate) -> Int{
```
Comment (alanscarpa) [inline twitterMini/NSDate+TimeFrom.swift]: add space between date:NSDate and everywhere else.

Comment (alanscarpa) [inline twitterMini/NSDate+TimeFrom.swift]: also add space between Int{   everywhere.

File: twitterMini/Services/TwitterAPIClient.swift [hunk 2]
```diff
@@ -23,12 +23,15 @@ class TwitterAPIClient {
...
     private var idForLoggedInUser: String?
+    var screenNameForLoggedInUser: String?
     
     func authorizeTwitter(completion: (Result<Void>) -> Void) {
         if let callbackURL = NSURL(string: self.callbackURL) {
             
             let success = { (credential: OAuthSwiftCredential, response: NSURLResponse?, parameters: [String:String]) in
                 self.idForLoggedInUser = parameters["user_id"]
+                self.screenNameForLoggedInUser = parameters["screen_name"]
+
```
Comment (alanscarpa) [inline twitterMini/Services/TwitterAPIClient.swift]: extra line

File: twitterMini/Services/TwitterDataManager.swift [hunk 3]
```diff
@@ -34,9 +49,11 @@ class TwitterDataManager {
     
     private let twitter = TwitterAPIClient()
     var statuses = [TwitterStatus]()
+    var userScreenName : String?
```
Comment (alanscarpa) [inline twitterMini/Services/TwitterDataManager.swift]: remove space Name :

File: twitterMini/Services/TwitterDataManager.swift [hunk 4]
```diff
@@ -34,9 +49,11 @@ class TwitterDataManager {
...
     
     func authorize(completion:(Result<Void>) -> Void) {
```
Comment (alanscarpa) [inline twitterMini/Services/TwitterDataManager.swift]: space after :

File: twitterMini/Services/TwitterDataManager.swift [hunk 5]
```diff
@@ -46,26 +63,43 @@ class TwitterDataManager {
...
-                            let user = TwitterUser(screenName: screenName, name: name, profileImageURL: profileImageURL)
-                            let status = TwitterStatus(createdAt: createdAt, text: text, user: user, likes: "\(likes)", retweets: "\(retweets)")
-                            self.statuses.append(status)
-                        }
-                    }
+                if let value = result.value,
+                    let rawStatuses = value {
+                    self.statuses = self.createStatuses(fromDictionaries: rawStatuses)
                 }
                 completion(.Success())
```
Comment (alanscarpa) [inline twitterMini/Services/TwitterDataManager.swift:70]: Does .Success need the () after it?  if not..remove

File: twitterMini/Services/TwitterDataManager.swift [hunk 6]
```diff
@@ -46,26 +63,43 @@ class TwitterDataManager {
...
+            if let createdAt = statusDictionary["created_at"] as? String,
+                let text = statusDictionary["text"] as? String,
+                let likes = statusDictionary["favorite_count"] as? Int,
+                let retweets = statusDictionary["retweet_count"] as? Int,
+                let userDictionary = statusDictionary["user"] as? [String: AnyObject],
+                let name = userDictionary["name"] as? String,
+                let screenName = userDictionary["screen_name"] as? String,
+                let profileImageURL = userDictionary["profile_image_url_https"] as? String {
+                
+                var mediaURL : String? = nil
```
Comment (alanscarpa) [inline twitterMini/Services/TwitterDataManager.swift]: var mediaURL: String?   is all you need.  it's set to nil by default when optional.

File: twitterMini/ViewControllers/HomeStatuses/TwitterTableViewCell.swift [hunk 7]
```diff
@@ -18,11 +19,12 @@ class TwitterTableViewCell : UITableViewCell {
...
     var status: TwitterStatus?{
         didSet{
-            if let status = status {
-                userNameLabel.text = status.user.name
-                userScreenNameLabel.text = "@\(status.user.screenName.lowercaseString)"
-//                createdAtLabel.text = status.createdAt // FIXME: just horrible
-                textContentLabel.text = status.text
+            if let status = self.status {
+                self.userNameLabel.text = status.user.name
+                self.userScreenNameLabel.text = "@\(status.user.screenName)"
```
Comment (alanscarpa) [inline twitterMini/ViewControllers/HomeStatuses/TwitterTableViewCell.swift]: reminds me of obj c days

**Comment by alanscarpa** [comment]
lookin' pretty good!

File: twitterMini/Services/TwitterDataManager.swift (hunk 5 above)
Comment (tomalley104) [inline twitterMini/Services/TwitterDataManager.swift:70]: it does 👎

File: twitterMini/Services/NSDate+TimeFrom.swift [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+//  twitterMini
+//
+//  Created by Tom O'Malley on 5/3/16.
+//  Copyright © 2016 intrepid. All rights reserved.
+//
+
+import Foundation
+
+extension NSDate {
+    func yearsFrom(date:NSDate) -> Int{
```
Comment (cprime) [inline twitterMini/Services/NSDate+TimeFrom.swift]: Spacing

File: twitterMini/ViewControllers/HomeStatuses/HomeStatusesTableViewController.swift [hunk 9]
```diff
@@ -11,18 +11,18 @@ import OAuthSwift
```
Comment (cprime) [inline twitterMini/ViewControllers/HomeStatuses/HomeStatusesTableViewController.swift:11]: Remove above .DS_Store files

**Comment by tomalley104** [comment]
@cprime squashed and merged. The code using Genome isn't 100% there so I'm putting it in it's own branch in case I decide to revisit it.

**Pull request merged**